src/pages/resultsView/coExpression/CoExpressionTab.tsx

```diff
@@ -90,53 +142,112 @@ export default class CoExpressionTab extends React.Component<ICoExpressionTabPro
+            this.props.store.coexpressionTabMolecularQueryProfiles.isComplete &&
+            this.props.store.coexpressionTabMolecularQueryProfiles.isComplete) {
+            const subjectOptionsGene = this.props.store.coexpressionTabMolecularSubjectProfilesGene.result.map(profile=>{
+                const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                return {
+                    label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                    value: profile.molecularProfileId
+                };
+            });
+            const subjectOptionsGeneset = this.props.store.coexpressionTabMolecularSubjectProfilesGeneset.result.map(profile=>{
+                const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                return {
+                    label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                    value: profile.molecularProfileId
+                };
+            });
+            const queryOptions = this.props.store.coexpressionTabMolecularQueryProfiles.result.map(profile=>{
+                    const profiledSampleCount = this.props.store.molecularProfileIdToProfiledSampleCount.result![profile.molecularProfileId];
+                    return {
+                        label: `${profile.name} (${profiledSampleCount} sample${profiledSampleCount !== 1 ? "s" : ""})`,
+                        value: profile.molecularProfileId
+                    };
+                });
             return (
-                <div
+                <div>
+                    <div
+                        style = {{
+                            display: "flex",
+                            alignItems: "center",
+                            padding: "5px"
+                        }}
+                    >
+                        <span>Subject Profile:</span>
+                        <div style={{display:"inline-block", width:376, marginLeft:4, marginRight:4, zIndex:15 /* so that on top when opened*/}}>
+                            <Select
+                                name="subject-profile-select"
+                                value={this.isSelectedGeneticEntityAGeneSet ? this.selectedSubjectProfileGeneset.molecularProfileId : this.selectedSubjectProfileGene.molecularProfileId}
+                                onChange={this.isSelectedGeneticEntityAGeneSet ? this.onSelectSubjectProfileGeneset : this.onSelectSubjectProfileGene}
+                                options={this.isSelectedGeneticEntityAGeneSet ? subjectOptionsGeneset : subjectOptionsGene}
+                                searchable={false}
+                                clearable={false}
+                                disabled={this.isSelectedGeneticEntityAGeneSet ? true : false}
```

Comment:
you dont need the ternary here, `this.isSelectedGeneticEntityAGeneset` is already boolean :), so just `disabled={this.isSelectedGeneticEntityAGeneset}`

Reply:
@adamabeshouse good catch 👍